cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -732,6 +732,11 @@ Result<Datum> IsNull(const Datum& arg, NullOptions options, ExecContext* ctx) {
   return CallFunction("is_null", {arg}, &options, ctx);
 }
 
+Result<Datum> Misbehave(const Datum& cond, const Datum& if_true, const Datum& if_false,
+                        ExecContext* ctx) {
+  return CallFunction("misbehave", {}, ctx);
+}
```

Comment:
These should not be added in the public APIs. Instead, these functions should only exist in the corresponding test file, IMHO.

Reply:
I've removed them again, also from the function registry. I'm now using a different way of calling the function through an executor, instead of going through `CallFunction`